Library/Homebrew/cleaner.rb

```diff
@@ -153,6 +154,48 @@ def clean_dir(directory)
     end
   end
 
+  sig { void }
+  def rewrite_pkgconfig
+    basepath = @formula.prefix.realpath
+    pc_files = %w[lib share].flat_map do |subdir|
+      pc_dir = basepath/subdir/"pkgconfig"
+      next [] if !pc_dir.exist? || @formula.skip_clean?(basepath/subdir) || @formula.skip_clean?(pc_dir)
+
+      pc_dir.glob("*.pc").reject { |pc_file| @formula.skip_clean?(pc_file) }
+    end
+    return if pc_files.empty?
+
+    # TODO: Add support for `brew unlink`-ed formulae and check on recursive dependencies
+    deps_pc_files = @formula.deps
+                            .filter_map { |dep| dep.to_formula if !dep.build? && !dep.test? }
+                            .select(&:keg_only?)
+                            .flat_map { |f| f.opt_prefix.glob("{lib,share}/pkgconfig/*.pc") }
+                            .to_h { |pc_file| [pc_file.basename(".pc").to_s.downcase, pc_file.to_s] }
```

Comment:
Need to figure out if there is a way to deal with whitespaces as this will break pkgconfig file. If I can't figure out a way, can just skip the modification if we detect whitespaces.

There is also an odd case of pkgconfig file backslash handling, e.g.
```
Cflags: -I/this/\
is/a/single/path
```
results in:
```console
❯ PKG_CONFIG_PATH=. pkg-config test --cflags
-I/this/is/a/single/path
```

May just ignore this for now until we find a scenario that needs it.